Source/Core/InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServer.h

```diff
@@ -0,0 +1,81 @@
+// Copyright 2019 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <chrono>
+
+#include "Common/Logging/Log.h"
+#include "Common/Matrix.h"
+#include "InputCommon/ControllerInterface/CemuHookUDPServer/CemuHookUDPServerProto.h"
+#include "InputCommon/ControllerInterface/ControllerInterface.h"
+#include "SFML/Network/UdpSocket.hpp"
+
+namespace ciface::CemuHookUDPServer
+{
+void Init();
+void PopulateDevices();
+void DeInit();
+void SaveSettings();
+
+class Device : public Core::Device
+{
+private:
+  template <class T>
```

Comment:
As far as I can tell, all of these internal classes aren't explicitly required in the interface, and can be moved into the cpp file to be hidden from the header.